linera-views/README.md

```diff
@@ -11,6 +11,134 @@ atomically persisted in storage).
+* Also, the list of the keys are provided in the lexicographic order.
+
+# The journaling.
+
+We want to make sure that all operations can be done on the database in a sure
+way. That is the operation have to be processed completely. The idea is to
+store the journal in the `[base_key 0 * * * *]` keys with `[* * * *]` the
+serialization of an `u32` entry. Once written, the journal is processed
+until all entries have been treated.
+Those keys are not overlaping with the other keys since their value is of the
+form `[base_key tag *]` with `tag >= MIN_VIEW_TAG` and currently we have
+`MIN_VIEW_TAG = 1`.
+
+
+## The `KeyValueStoreClientBigValue` type.
+
+Each key value store client has a maximal size of its values (named `MAX_VALUE_SIZE`
+in the code). If we want to have unlimited size, then we can build a
+key-value store client with unlimited size.
+
+The built key-value store client has `MAX_VALUE_SIZE` set to its maximum possible
+value. This forces us to split the value into several smaller values. One key
+such example is `DynamoDb` which has a limit of 400kB on its value size.
+
+Design is the following:
+* For every `key` we build several corresponding keys `[key * * * *]` that contain
+each a piece of the full data set.
+The `[* * * *]` is the serialization of the index which is `u32` so that
+the order of the serialization is the same as the order of the `u32` entries.
+* For the first key, that is `[key 0 0 0 0]` the information is `[* * * * value_0]`
+with `[* * * *]` being the serialization of the `u32` count of the number of blocks
+and `value_0` the first block of the value. For the other keys `[key * * * *]` the
+corresponding value is `value_k` with `value_k` the k-th entry in the value. The full
+value is reconstructed as `value = [value_0 value_1 .... value_N]`
+The number of blocks is always non-zero, even if the value is empty.
+
+The effective working of the system relies on several design choices.
+* In `find_keys_by_prefix` and similar, the keys are
+returned in lexicographic order.
+* The count and indices are serialized as `u32` and the serialization is done
+so that the `u32` ordering is the same as the lexicographic ordering. So, we
+get first the first key (and so the count) and then the segment keys one by
+one in the correct order.
+* Suppose that we have written a `(key, value_1)` with several segments and then
+we replace by a `(key, value_2)` which has fewer segments.
+Our choice is to leave the old segments in the database. Same with delete:
+we just delete the first key.
+* We avoid the overflowing of the system by the following trick: When we delete
+a view, we do a delete by prefix, which thus deletes the old segments if present.
+* In order to have a segment that deletes all the unused segments, every write needs
+to have a read just before which is expensive.
+
+key difficulty of this modelization is that while deleting a key of the form
+`[key 0 0]` we would also delete segments of the key of the form `key`.
```

Comment:
```suggestion
A difficulty with this approach is that while deleting a key of the form
`[prefix 0 0]` we would also delete segments of a value stored under the key
`prefix`.
```

…but maybe that's too much information for a README? Otherwise maybe it needs an even more detailed explanation.